phpini/phpini-lib.pl

```diff
@@ -992,29 +1014,47 @@ sub list_all_php_version_packages
 my @rv = map { $_->{'name'} }
 	     &list_all_php_module_packages($pkg->{'name'});
 my $base = $pkg->{'name'};
-$base =~ s/-php$//;
-my @poss = ( @modpkgs, $base."-php", $base."-runtime", $base );
+$base =~ s/-php-common$//;
+$base =~ s/-common$//;
+my @poss = ( $base."-runtime", $base );
 foreach my $p (@poss) {
-	my @info = &software::package_info($p, $pkg->{'ver'});
+	my @info = &software::package_info($p);
 	next if (!@info);
 	push(@rv, $p);
 	}
 return @rv;
 }
 
-# delete_php_base_package(&package)
+# extend_installable_php_packages(&packages)
+# Given a list of PHP packages to install, extends them to include packages
+# that are also has to be installed, such as -cli or -fpm
+sub extend_installable_php_packages
+{
+my ($pkgs) = @_;
+my @extra = ('cli', 'fpm');
+foreach my $pkg (@$pkgs) {
+	if ($pkg->{'name'} =~ /-common$/) {
+		$pkg->{'name'} .= ' ' . join(' ',
```

Comment:
I don't think modifying the `name` field to turn it into a list of package names is a clean approach.

Reply:
Well, what is the other better suggestion? We need to specifically extend it before making an install.

Reply:
I'm not sure, you wrote the code! :-) 

But in this package object, `name` is a field that just contains the package name. It's not reasonable to redefine it to be something else for convenience later in the code..

Reply:
Okay, I simplified the code to avoid using the newer `/r` flag, to have a potential support for systems with Perl 5.10...

Reply:
Instead of modifying the list passed in, this function should return a new list contaiing whatever the caller needs. That's where it would make sense to list multiple packages..

Also, resist the temption put everything on one line line this. It's a lot more readable to use temporary variables.

Reply:
> Instead of modifying the list passed in, this function should return a new list contaiing whatever the caller needs. That's where it would make sense to list multiple packages..

Agreed! Thanks for the suggestion. Fixed here https://github.com/webmin/webmin/pull/2483/commits/92dc310ce0ee39c491f7bd931d9ed68b0b567af3.



> Also, resist the temptation put everything on one line line this. It's a lot more readable to use temporary variables.

Yeah, I like one-liners.